### PR TITLE
Feat/UI t 23 profile

### PR DESCRIPTION
**Pull request by samvanhoeyicapps**

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -323,7 +323,7 @@ PODS:
     - react-native-config/App (= 1.4.11)
   - react-native-config/App (1.4.11):
     - React-Core
-  - react-native-flipper (0.163.0):
+  - react-native-flipper (0.174.0):
```
Comment (hstandaert) [inline ios/Podfile.lock:326]: Don't use this version in the podfile 🙃

File: src/_components/linkList/LinkList.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+type TProps = {
+  items: TLinkListItem[];
+};
+
+const LinkList: FC<TProps> = ({ items, ...props }) => {
+  function renderItem(item: TLinkListItem) {
+    return (
+      <Styled.LinkItem key={item.label}>
+        <Icon color={item.iconColor} name={item.iconName} />
+        {item.href ? (
```
Comment (hstandaert) [inline src/_components/linkList/LinkList.tsx:25]: There's no added value for the if statement.. I think you can just pass in both the `href` and `onPress` props, since the typing matches the type of the `Button` props.

File: src/_components/linkList/LinkList.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  label: string;
+  labelColor: keyof Theme['colors'];
+} & TLinkTypes;
+
+type TProps = {
+  items: TLinkListItem[];
+};
+
+const LinkList: FC<TProps> = ({ items, ...props }) => {
+  function renderItem(item: TLinkListItem) {
```
Comment (hstandaert) [inline src/_components/linkList/LinkList.tsx:21]: Preferably, this is a separate component (in a separate file) 😇

File: src/_components/typography/style.ts [hunk 4]
```diff
@@ -7,25 +7,6 @@ export const Typography = styled.Text<TTypographyProps>`
...
-        return '27px';
-
-      case 'xlarge':
-        return '30px';
-
-      case 'xxlarge':
-        return '36px';
-    }
-  }};
-
```
Comment (hstandaert) [inline src/_components/typography/style.ts:28]: Why is this removed? 🤔

File: src/_routing/_components/ProfileNavigator.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+
+import { theme } from '../../_styles/theme';
+import History from '../../history/History';
+import Profile from '../../profile/Profile';
+
+export type TProfileParams = {
+  History: undefined;
+  Profiel: undefined;
+};
```
Comment (hstandaert) [inline src/_routing/_components/ProfileNavigator.tsx:11]: This should be in line with the other navigators: 

```suggestion
export type TProfileRoutes = 'Profile' | 'History';
export type TProfileParams = Record<TProfileRoutes, undefined>;
```

Furthermore, there's a dutch word in there 🚨

File: src/_styles/theme.ts [hunk 6]
```diff
@@ -1,14 +1,23 @@
...
+    lightBlue: '#90D9DA',
+    lightGreen: '#81D5BE',
+    lightGrey: '#CFD3D3',
+
     primary: '#39ac8d',
     primaryDark: '#127173',
+    red: '#F82E58',
     secondary: '#149773',
     secondaryDark: '#118363',
+    teal: '#1AA5A8',
```
Comment (hstandaert) [inline src/_styles/theme.ts:20]: Might be nitpicking, but I find it more readable if you use lowercase letters in HEX values

File: src/_translations/locales/nl.json [hunk 7]
```diff
@@ -26,6 +26,18 @@
     "WHERE_TO_BUY": "Waar koop ik een UiTPAS?",
     "LOGIN": "Aanmelden"
   },
+  "PROFILE": {
+    "POINTS": "punten",
+    "HELLO": "Dag",
```
Comment (hstandaert) [inline src/_translations/locales/nl.json:31]: More futureproof to use interpolation here.

```suggestion
    "HELLO": "Dag {{name}}",
```

File: src/profile/MIANotification/MIANotification.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import * as Styled from './style';
+
+type TProps = {
+  socialTariffInfo: TCardSystemMembership['socialTariff'];
+};
+
+const MIANotification: FC<TProps> = ({ socialTariffInfo }) => {
+  return (
+    <Styled.NotificationContainer>
+      <Styled.CloseButton name="KansenTarief" size={25} />
```
Comment (hstandaert) [inline src/profile/MIANotification/MIANotification.tsx:15]: Dutch 🚨

File: src/profile/MIANotification/MIANotification.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+      <Styled.MIAText color="darkGreen">{i18n.t('PROFILE.MIA')}</Styled.MIAText>
+      <Styled.ExpirationText color="darkGreen" size="small">{`${i18n.t('PROFILE.VALID_UNTIL')} ${formatDate(
+        new Date(socialTariffInfo?.endDate),
```
Comment (hstandaert) [inline src/profile/MIANotification/MIANotification.tsx:18]: The optional chaining operator indicates that `socialTariffInfo` is possibly undefined. But it's a required prop. 🤔 

Is there a use case where `socialTariffInfo?.endDate` is emtpy? In that case the `new Date(undefined)` will display 'Invalid Date'. Might need some error handling..

File: src/profile/UitpasCard/UitpasCard.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  const activeUitpasCards = passHolder.cardSystemMemberships.filter(card => card.status === 'ACTIVE' && card.uitpasNumber);
+  const [firstActiveCard] = activeUitpasCards;
+
+  function renderCards() {
+    if (activeUitpasCards.length === 1) {
+      return firstActiveCard.cardSystem.name;
+    } else {
+      return `${firstActiveCard.cardSystem.name}, ...`;
+    }
+  }
```
Comment (hstandaert) [inline src/profile/UitpasCard/UitpasCard.tsx:25]: You could do this inline like:

```ts
{firstActiveCard.cardSystem.name}
{activeUitpasCards.length > 1 && ", ..."}
```

File: src/profile/_queries/useGetHistory.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    retry: (_, error) => {
+      if (error.status === HttpStatus.NotFound) {
+        return false;
+      }
+
+      setRetryCount(retryCount + 1);
+      return retryCount < 6;
+    },
+  });
+}
```
Comment (hstandaert) [inline src/profile/_queries/useGetHistory.ts:28]: I think this was not meant to be committed yet? 🤔

**Review by hstandaert — CHANGES_REQUESTED** [review]

File: src/_components/linkList/LinkList.tsx (hunk 2 above)
Comment (samvanhoeyicapps) [reply src/_components/linkList/LinkList.tsx:25]: no 😢

File: src/_routing/_components/ProfileNavigator.tsx (hunk 5 above)
Comment (samvanhoeyicapps) [reply src/_routing/_components/ProfileNavigator.tsx:11]: opgelost in volgende pr

File: src/_styles/theme.ts (hunk 6 above)
Comment (samvanhoeyicapps) [reply src/_styles/theme.ts:20]: indeed nitpicking 😉

File: src/_translations/locales/nl.json (hunk 7 above)
Comment (samvanhoeyicapps) [reply src/_translations/locales/nl.json:31]: fixed in next PR

File: ios/Podfile.lock (hunk 1 above)
Comment (samvanhoeyicapps) [reply ios/Podfile.lock:326]: fixed in next PR

**Review by hstandaert — DISMISSED** [review]
